internal/fs/fs.go

```diff
@@ -1619,26 +1618,19 @@ func (fs *fileSystem) createLocalFile(
 		}
 	}()
 
-	fs.mu.Lock()
-
 	fullName := inode.NewFileName(parent.Name(), name)
 	child, ok := fs.localFileInodes[fullName]
 	if !ok {
-		var result *inode.Core
-		result, err = parent.CreateLocalChildFile(name)
-		if err != nil {
-			return
-		}
-
-		child = fs.mintInode(*result)
-		fs.localFileInodes[child.Name()] = child
-
-		// Empty file is created to be able to set attributes on the file.
-		fileInode := child.(*inode.FileInode)
-		err = fileInode.CreateEmptyTempFile()
-		if err != nil {
-			return
-		}
+		parent.Lock()
```

Comment:
This doesn't follow the lock ordering. parent.Lock() should be acquired before fs.Lock()

Reply:
I've already added a test for it. Also fixed the locking.